source/server/admin/admin.cc

```diff
@@ -469,9 +484,16 @@ void AdminImpl::addAllConfigToDump(envoy::admin::v3::ConfigDump& dump,
 
 absl::optional<std::pair<Http::Code, std::string>>
 AdminImpl::addResourceToDump(envoy::admin::v3::ConfigDump& dump,
-                             const absl::optional<std::string>& mask,
-                             const std::string& resource) const {
-  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+                             const absl::optional<std::string>& mask, const std::string& resource,
+                             const bool include_eds) const {
+  Envoy::Server::ConfigTracker::CbsMap callbacks_map = std::move(config_tracker_.getCallbacksMap());
+  if (include_eds) {
+    if (!server_.clusterManager().clusters().empty()) {
+      callbacks_map.emplace("endpoint", [this] { return dumpEndpointConfigs(); });
+    }
+  }
+
+  for (const auto& key_callback_pair : callbacks_map) {
```

Comment:
should we also honor the resource for eds dump?

Reply:
Since the eds dump has been already added in the `callbacks_map` via `callbacks_map.emplace("endpoint", [this] { return dumpEndpointConfigs(); });`, I think the resource of it will be automatically filtered as others in this for loop?

Reply:
Cool. can we add test to cover this?

Reply:
Added test `ConfigDumpWithEndpointFiltersByResource`